### PR TITLE
ecsk: init at 0.9.3

### PR DESCRIPTION
**Pull request by whtsht**
[ecsk](https://github.com/yukiarrr/ecsk) is a CLI tool that you can interactively call Amazon ECS APIs (run-task, execute-command, stop-task), copy files between ECS and local, and view logs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ec/ecsk/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  vendorHash = "sha256-Eyqpc7GyG/7u/I4tStADQikxcbIatjeAJN9wUDgzdFY=";
+
+  buildPhase = ''
+    go build -o $out/bin/ecsk ./cmd/ecsk
+  '';
+
+  meta = with lib; {
+    description = "CLI tool that you can interactively call Amazon ECS APIs, copy files between ECS and local, and view logs.";
```
Comment (Aleksanaa) [inline pkgs/by-name/ec/ecsk/package.nix:26]:
```suggestion
    description = "Interactively call Amazon ECS APIs, copy files between ECS and local, and view logs";
```

File: pkgs/by-name/ec/ecsk/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    license = licenses.mit;
+    mainProgram = "ecsk";
+    maintainers = with maintainers; [ whtsht ];
+  };
```
Comment (Aleksanaa) [inline pkgs/by-name/ec/ecsk/package.nix:30]:
```suggestion
  meta = {
    description = "CLI tool that you can interactively call Amazon ECS APIs, copy files between ECS and local, and view logs.";
    license = lib.licenses.mit;
    mainProgram = "ecsk";
    maintainers = with lib.maintainers; [ whtsht ];
  };
```

Comment (Aleksanaa) [reply pkgs/by-name/ec/ecsk/package.nix:30]: Also missing `homepage`

File: pkgs/by-name/ec/ecsk/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+buildGoModule rec {
+  pname = "ecsk";
+  version = "0.9.3";
+
+  src = fetchgit {
+    url = "https://github.com/yukiarrr/ecsk.git";
+    rev = "v${version}";
+    sha256 = "sha256-1nrV7NslOIXQDHsc7c5YfaWhoJ8kfkEQseoVVeENrHM=";
+    fetchSubmodules = true;
+  };
```
Comment (Aleksanaa) [inline pkgs/by-name/ec/ecsk/package.nix:17]:
```suggestion
  src = fetchFromGitHub {
    owner = "yukiarrr";
    repo = "ecsk";
    rev = "refs/tags/v${version}";
    hash = "";
    fetchSubmodules = true;
  };
```

refetch hash.

File: pkgs/by-name/ec/ecsk/package.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  vendorHash = "sha256-Eyqpc7GyG/7u/I4tStADQikxcbIatjeAJN9wUDgzdFY=";
+
+  buildPhase = ''
+    go build -o $out/bin/ecsk ./cmd/ecsk
+  '';
```
Comment (Aleksanaa) [inline pkgs/by-name/ec/ecsk/package.nix:23]:
```suggestion
  subPackages = [ "cmd/esck" ];
```

Just guessing.

File: pkgs/by-name/ec/ecsk/package.nix (hunk 2 above)
Comment (whtsht) [reply pkgs/by-name/ec/ecsk/package.nix]: Added Github link as homepage

**Review by NotAShelf — APPROVED** [review]
LGTM

**Pull request merged**